source/FreeRTOS_Routing.c

```diff
@@ -615,84 +627,74 @@ struct xIPv6_Couple
                                          const IPv6_Address_t * pxDNSServerAddress,
                                          const uint8_t ucMACAddress[ ipMAC_ADDRESS_LENGTH_BYTES ] )
         {
-            #if ( ipconfigUSE_DHCPv6 != 0 ) || ( ipconfigUSE_RA != 0 )
-                /* Assume that the endpoint has a static address assignment. */
-                BaseType_t xIsStatic = pdTRUE;
-            #endif
-            configASSERT( pxIPAddress != NULL );
-            configASSERT( ucMACAddress != NULL );
-            configASSERT( pxEndPoint != NULL );
-
-            ( void ) memset( pxEndPoint, 0, sizeof( *pxEndPoint ) );
-
-            pxEndPoint->bits.bIPv6 = pdTRUE_UNSIGNED;
-
-            pxEndPoint->ipv6_settings.uxPrefixLength = uxPrefixLength;
-
-            if( pxGatewayAddress != NULL )
+            if( ( pxNetworkInterface == NULL ) ||
+                ( pxEndPoint == NULL ) ||
+                ( pxIPAddress == NULL ) ||
+                ( ucMACAddress == NULL ) )
```

Comment:
In my opinion asserts are more easier to set up and helpful while debugging than `printf`s.

Reply:
Because this is a public API, I believe less assertion the better.